### ISSUE TITLE
Fastlane.swift keeps Gitlab Runner job running forever

### ISSUE THREAD

**Issue by AhmedOS** [fastlane/fastlane#15955]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

- Fastlane.swift keeps Gitlab Runner job running forever and prevents it from finishing, even that Fastlane doesn't produce any errors, also, runner continue to execute following scripts after Fastlane one, if there is any.
- Fastlane.swift works fine by itself if I executed `bundle exec fastlane build` through terminal.
- I have removed Fastlane.swift from repo and installed Fastlane ruby version, and everything goes fine with Gitlab Runner.

#### - `gitlab-ci.yml` file:
```
stages:
  - build

variables:
  LC_ALL: "en_US.UTF-8"
  LANG: "en_US.UTF-8"
  GIT_STRATEGY: clone

before_script:
   - gem install bundler
   - bundle install

build:
  stage: build
  script:
    - bundle exec fastlane build --verbose
```

#### - Gitlab Runner job doesn't finish:
![](https://i.imgur.com/nncRl4z.png)

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.6                                     |
| Ruby                        | 2.6.5                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.21.0 (Apple Git-122.2)        |
| Installation Source         | ~/.rvm/gems/ruby-2.6.5/bin/fastlane         |
| Host                        | Mac OS X 10.14.6 (18G103)                   |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.6.5/lib                |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.3                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile.swift`</summary>

```ruby
// This file contains the fastlane.tools configuration
// You can find the documentation at https://docs.fastlane.tools
//
// For a list of all available actions, check out
//
//     https://docs.fastlane.tools/actions
//

import Foundation

class Fastfile: LaneFile {
    
    func buildLane() {
        //linting()
        //handleArtifacts()
    }
    
    func linting() {
        swiftlint(
          mode: "lint",
          outputFile: "swiftlint.result.json",
          ignoreExitStatus: true
        )
    }
    
    func handleArtifacts() {
        copyArtifacts(keepOriginal: false, targetPath: "~/Desktop/artifacts", artifacts: ["swiftlint.result.json"], failOnMissing: false)
    }

}

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.141.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 3.0.2        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.14.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.1        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.0        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.72.0       |
| faraday                   | 0.17.3       |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 1.17.3       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.7        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.14.1       |
| signet                    | 0.12.0       |
| memoist                   | 0.16.2       |
| os                        | 1.0.1        |
| googleauth                | 0.10.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.36.4       |
| google-cloud-env          | 1.3.0        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-core         | 1.5.0        |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.25.1       |
| emoji_regex               | 1.0.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| json                      | 2.3.0        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
</details>

*generated on:* **2020-01-30**
</details>

**Comment by joshdholtz** [comment]
Oh yikes! This sounds like there may be an issue with the socket that Swift and Ruby communicate over that is never finishing up 🤔 This one is either an easy fix or not an easy fix 😱

**Comment by Balake** [comment]
Having the same issue also after updating everything to latest

**Comment by joshdholtz** [comment]
@Balake With Fastlane.swift? Or Ruby fastlane?

**Comment by Balake** [comment]
@joshdholtz Fastlane.swift

**Comment by joshdholtz** [comment]
Thanks! Just wanted to make sure 😇 I'll add this into my queue. I will need figure out how to spin up a gitlab runner and all that jazz so this might not be a quick fix. But I am really intrigued in what could be causing this 🤔

**Comment by Balake** [comment]
@joshdholtz not sure if this is a fastlane issue anymore... Some more context: I had rvm installed to pull the latest ruby. This introduced some issues with gitlab runner as [here](https://gitlab.com/gitlab-org/gitlab-runner/issues/114#note_4399113). I fixed that, but then ran into this issue. Finally, after exhausting too many combinations of gitlab-runner and fastlane, I decided to just remove rvm (`rvm implode`). This has fixed my issue.
